contracts/test/foreign-proxy-with-appeals.js

```diff
@@ -0,0 +1,798 @@
+  await foreignProxy.setHomeProxy(homeProxy.address, homeChainID);
+  await foreignProxy.connect(requester).requestArbitration(questionID, answer, { value: arbitrationCost });
+
+  await homeProxy.handleNotifiedRequest(questionID);
+
+  await arbitrator.giveAppealableRuling(2, 21, appealCost, appealTimeOut);
+
+  await time.increase(appealTimeOut / 2 + 1);
+
+  // Loser.
+  await expect(
+    foreignProxy.connect(crowdfunder1).fundAppeal(arbitrationID, 533, { value: appealCost })
+  ).to.be.revertedWith("Appeal period is over for loser.");
+
+  // Adding another half will cover the whole period.
+  await time.increase(appealTimeOut / 2 + 1);
+
+  // Winner.
+  await expect(
+    foreignProxy.connect(crowdfunder1).fundAppeal(arbitrationID, 20, { value: appealCost })
+  ).to.be.revertedWith("Appeal period is over.");
+});
+
+it("Should correctly withdraw appeal fees if a dispute had winner/loser", async () => {
+  let oldBalance1;
+  let oldBalance2;
+  let newBalance;
+  let newBalance1;
+  let newBalance2;
+  const requesterAddress = await requester.getAddress();
+  const crowdfunder1Address = await crowdfunder1.getAddress();
+  const crowdfunder2Address = await crowdfunder2.getAddress();
+
+  await foreignProxy.setHomeProxy(homeProxy.address, homeChainID);
+  await foreignProxy.connect(requester).requestArbitration(questionID, answer, { value: arbitrationCost });
+
+  await homeProxy.handleNotifiedRequest(questionID);
+
+  await arbitrator.giveAppealableRuling(2, 5, appealCost, appealTimeOut);
+
+  // LoserFee = 8500, WinnerFee = 6500. AppealCost = 5000.
+  // 0 Round.
+  await foreignProxy.connect(requester).fundAppeal(arbitrationID, 50, { value: 4000 });
+  await foreignProxy.connect(crowdfunder1).fundAppeal(arbitrationID, 50, { value: oneETH });
+
+  await foreignProxy.connect(crowdfunder2).fundAppeal(arbitrationID, 4, { value: 6000 });
+  await foreignProxy.connect(crowdfunder1).fundAppeal(arbitrationID, 4, { value: 6000 });
+
+  await arbitrator.giveAppealableRuling(2, 5, appealCost, appealTimeOut);
+
+  // 1 Round.
+  await foreignProxy.connect(requester).fundAppeal(arbitrationID, 44, { value: 500 });
+  await foreignProxy.connect(crowdfunder1).fundAppeal(arbitrationID, 44, { value: 8000 });
+
+  await foreignProxy.connect(crowdfunder2).fundAppeal(arbitrationID, 4, { value: 20000 });
+
+  await arbitrator.giveAppealableRuling(2, 5, appealCost, appealTimeOut);
+
+  // 2 Round.
+  // Partially funded side should be reimbursed.
+  await foreignProxy.connect(requester).fundAppeal(arbitrationID, 41, { value: 8499 });
+
+  // Winner doesn't have to fund appeal in this case but let's check if it causes unexpected behaviour.
+  await foreignProxy.connect(crowdfunder2).fundAppeal(arbitrationID, 4, { value: oneETH });
+
+  await time.increase(appealTimeOut + 1);
+
+  await expect(foreignProxy.withdrawFeesAndRewards(arbitrationID, requesterAddress, 0, 50)).to.be.revertedWith(
+    "Dispute not resolved"
+  );
```

Comment:
Should be in its own test case "It should not allow to withdraw fees and rewards when the dispute is not resolved".